executor/src/lib.rs

```diff
@@ -626,3 +633,787 @@ impl ExecutedTrees {
+                hash_map::Entry::Occupied(mut entry) => {
+                    let account_btree = entry.get_mut();
+                    // TODO(gzh): we check account creation here for now. Will remove it once we
+                    // have a better way.
+                    if account_btree.is_empty() {
+                        num_accounts_created += 1;
+                    }
+                    Self::update_account_btree(account_btree, path, write_op);
+                }
+                hash_map::Entry::Vacant(entry) => {
+                    // Before writing to an account, VM should always read that account. So we
+                    // should not reach this code path. The exception is genesis transaction (and
+                    // maybe other FTVM transactions).
+                    match transaction.as_signed_user_txn()?.payload() {
+                        TransactionPayload::Program
+                        | TransactionPayload::Module(_)
+                        | TransactionPayload::Script(_) => {
+                            bail!("Write set should be a subset of read set.")
+                        }
+                        TransactionPayload::WriteSet(_) => (),
+                    }
+
+                    let mut account_btree = BTreeMap::new();
+                    Self::update_account_btree(&mut account_btree, path, write_op);
+                    entry.insert(account_btree);
+                }
+            }
+            addrs.insert(address);
+        }
+
+        for addr in addrs {
+            let account_btree = account_to_btree.get(&addr).expect("Address should exist.");
+            let account_blob = AccountStateBlob::try_from(account_btree)?;
+            updated_blobs.insert(addr, account_blob);
+        }
+        let state_tree = Arc::new(
+            previous_state_tree
+                .update(
+                    updated_blobs
+                        .iter()
+                        .map(|(addr, value)| (addr.hash(), value.clone()))
+                        .collect(),
+                    proof_reader,
+                )
+                .expect("Failed to update state tree."),
+        );
+
+        Ok((updated_blobs, state_tree, num_accounts_created))
+    }
+
+    fn update_account_btree(
+        account_btree: &mut BTreeMap<Vec<u8>, Vec<u8>>,
+        path: Vec<u8>,
+        write_op: WriteOp,
+    ) {
+        match write_op {
+            WriteOp::Value(new_value) => account_btree.insert(path, new_value),
+            WriteOp::Deletion => account_btree.remove(&path),
+        };
+    }
+}
+
+struct ProofReader {
+    account_to_proof: HashMap<HashValue, SparseMerkleProof>,
+}
+
+impl ProofReader {
+    fn new(account_to_proof: HashMap<HashValue, SparseMerkleProof>) -> Self {
+        ProofReader { account_to_proof }
+    }
+}
+
+impl ProofRead for ProofReader {
+    fn get_proof(&self, key: HashValue) -> Option<&SparseMerkleProof> {
+        self.account_to_proof.get(&key)
+    }
+}
```

Comment:
what's the value in this snippet of code?

Reply:
This is used by line 861